### PR TITLE
fix: allign set_target_temperature signatures (part2)

### PR DESCRIPTION
**Pull request by chemelli74**
## Summary by CodeRabbit

- **New Features**
	- Introduced the ability to set target temperature for Midea FB devices with optional mode and zone parameters.

- **Improvements**
	- Enhanced Midea C3 device temperature setting functionality by adding a default value for the zone parameter.

### CODE REVIEW

**Review by rokam — APPROVED** [review]

File: midealocal/devices/c3/__init__.py [hunk 1]
```diff
@@ -290,11 +290,11 @@ def set_target_temperature(
         self,
         target_temperature: float,
         mode: int | None,
-        zone: int,
+        zone: int | None = None,
     ) -> None:
         """Midea C3 device set target temperature."""
         message = self.make_message_set()
-        if self._attributes[DeviceAttributes.zone_temp_type][zone]:
+        if zone and self._attributes[DeviceAttributes.zone_temp_type][zone]:
```
Comment (Necroneco) [inline midealocal/devices/c3/__init__.py:297]: What happens if zone == 0?

File: midealocal/devices/fb/__init__.py [hunk 2]
```diff
@@ -112,6 +112,14 @@ def set_attribute(self, attr: str, value: str | int | bool) -> None:
             setattr(message, str(attr), value)
         self.build_send(message)
 
+    def set_target_temperature(
```
Comment (Necroneco) [inline midealocal/devices/fb/__init__.py:115]: Why add `set_target_temperature` to FB device?

Comment (chemelli74) [reply midealocal/devices/fb/__init__.py:115]: BEacuse is needed on integration side as for this device we use directly set_attribute which has the wrong params

File: midealocal/devices/c3/__init__.py (hunk 1 above)
Comment (chemelli74) [reply midealocal/devices/c3/__init__.py:297]: is 0 a valid value ? I think zone is >=1 or None

Comment (Necroneco) [reply midealocal/devices/c3/__init__.py:297]: Obviously `zone` is 0 or 1

```python
            message.zone_target_temp[zone] = target_temperature
```

```python
            if zone == 0:
                message.zone1_power = True
            else:
                message.zone2_power = True
```

```python
        zone1_target_temp = int(self.zone_target_temp[0])
        zone2_target_temp = int(self.zone_target_temp[1])
```

Comment (chemelli74) [reply midealocal/devices/c3/__init__.py:297]: > Obviously `zone` is 0 or 1

or None ;-)

Comment (chemelli74) [reply midealocal/devices/c3/__init__.py:297]: Is enough to change the line to "if zone is not None..."

Comment (chemelli74) [reply midealocal/devices/c3/__init__.py:297]: #180

File: midealocal/devices/fb/__init__.py (hunk 2 above)
Comment (chemelli74) [reply midealocal/devices/fb/__init__.py:115]: #179

File: midealocal/devices/c3/__init__.py (hunk 1 above)
Comment (Necroneco) [reply midealocal/devices/c3/__init__.py:297]: `None` is not valid value, maybe should raise an error

**Pull request merged**